### PR TITLE
really drop python<=3.7 support

### PR DESCRIPTION
**Pull request by kloczek**
## Change Summary
Filter all code over `pyupgrade --py38`.

## Related issue number
N/A

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by Viicos** [comment]
I believe the v1 code should be left as is

**Comment by kloczek** [comment]
> Indeed, @Viicos is right! Let's leave the V1 code as is. Happy to review again once that's changed - thanks!

OK .. however I'm not sure about you are talking about so please close this PR and to it by yourself or merge than correct what still should be rolled back 😋 
If it anything wrong what have been filtered incorrectly by `pyupgrade` please report that on https://github.com/asottile/pyupgrade

**Comment by Viicos** [comment]
Basically the `v1` folder is a port of Pydantic V1 so that even in Pydantic V2, the v1 code can be imported under `pydantic.v1`. But this code should be identical to what is available on the [`1.x`](https://github.com/pydantic/pydantic/tree/1.10.X-fixes) branch.

File: pydantic/types.py [hunk 1]
```diff
@@ -2425,7 +2425,7 @@ class Model(BaseModel):
     def __get_pydantic_core_schema__(self, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
         return core_schema.with_info_after_validator_function(
             function=self.decode_str,
-            schema=super(EncodedStr, self).__get_pydantic_core_schema__(source=source, handler=handler),  # noqa: UP008
+            schema=super().__get_pydantic_core_schema__(source=source, handler=handler),  # noqa: UP008
```
Comment (Viicos) [inline pydantic/types.py:2428]: The noqa comment was actually there to prevent the `super()` change introduced. I'm not sure if it was intentional?

Comment (kloczek) [reply pydantic/types.py:2428]: So just do not filter that file only in corrected PR? 🤔

Comment (Viicos) [reply pydantic/types.py:2428]: Confirmed from this comment: https://github.com/pydantic/pydantic/pull/7396#issuecomment-1727348592

This one should be left as is.

So basically only the changes in `tests/benchmarks` can be kept :)

Comment (sydney-runkle) [reply pydantic/types.py]: @kloczek,

Go for it. Looks like there are a few linting issues as well...

Comment (Viicos) [reply pydantic/types.py]: > OK I've rediffed that to modify only tests/ but I see pydantic/types.py can be updated as well.

No as I said the explicit `super` call is required as mentioned in the following comment: https://github.com/pydantic/pydantic/pull/7396#issuecomment-1727348592

Comment (sydney-runkle) [reply pydantic/types.py]: So @kloczek,

Just the benchmarks changes should do

**Review by sydney-runkle — CHANGES_REQUESTED** [review]
Could you please remove the unused `typing.Dict` and `typing.Optional` imports? That should fix the pipeline for good!

**Comment by kloczek** [comment]
> Could you please remove the unused `typing.Dict` and `typing.Optional` imports? That should fix the pipeline for good!

In updated PR are only changes in tests/ and I see in that new diff any fiddling around imports 🤔

**Comment by sydney-runkle** [comment]
> In updated PR are only changes in tests/ and I see in that new diff any fiddling around imports 🤔

All good, I made the fixes. Thanks!

**Review by sydney-runkle — APPROVED** [review]

**Comment by kloczek** [comment]
OK now I see about what it was 😋 
How did you font that? Using ruff? 🤔

**Comment by sydney-runkle** [comment]
@kloczek,

Yep, you can use `ruff`. The issue was also reported in our `lint` jobs

**Comment by kloczek** [comment]
Thank you because I had again I've had opportunity to learned few small new details 👍 😄

**Pull request merged**